v2/bundleworkload/option.go

```diff
@@ -0,0 +1,37 @@
+package bundleworkload
+
+import (
+	"github.com/spiffe/go-spiffe/v2/logger"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+)
+
+// Option is used to provide additional options to the bundle workload.
+type Option interface{}
```

Comment:
Regarding options, the convention I went with was:
1) `Option` if the package did one thing and was unlikely going to need multiple option types
2) `Option` if the package did many things but the `Option` were general purpose and applied to other operations as well, e.g. grpcworkload.Option's can be used as `DialOption` and `ServerOption`, since they are common to both
3) `[PREFIXED]Option` when the package had more than one kind of operation needing distinct options (which I thought bundleendpoint would do, but it ended up not needed options except for Fetch)

Reply:
Regarding package names, I'm not in love with the current set of names. I would however detest package names like `jwt`, `x509`, `tls`, `bundle`, `svid`, etc, because those either 1) conflict with the stdlib and are likely to be used in the same packages together, expecially `x509`, or 2) are super useful variable names, e.g.
```
bundle, err := spiffebundle.Parse(blah)
```
 is much more attractive to me than
```
b, err := bundle.Parse(blah)
```
or heaven forbid:
```
bundle, err := bundle.Parse(blah)
```

Whatever we come up with, I don't want people to have to name their imports on a frequent basis. That's more smelly to me than a prefix.

Reply:
agree with `bundle` alone or another cases will result in problems when importing or using them

Reply:
it will be useful if we select an standard (and that is what we are trying to do), so people using it  can easly understarnd what each thing is, I'm still worried about whay I preffer for example:
workloadjwt or jwtworkload... 
but if we are going to use things like spiffebundle, we must do the same thing in all places.
If I put it in code it will looks like:

```
option := bundleWorkload.WithWorkloadAPIOptions(option1, option2)
```

vs

```
option := workloadbundle.WithWorkloadApiOptions(option1, option2)
```

too much `workload` in naming if you ask me, 
and in code I prefer to see `bundleWorkload` that looks better, it is hard to see we are talking about bundle if it is in middle